File: lib/src/data/use_cases/log.dart
import 'package:functionally/reader_io.dart';
import 'package:logging/logging.dart';

import '../../application/commands/log.dart';

final LogCommandReader<Logger> log = (type, message) => ReaderIO(
      (logger) => () => switch (type) {
            LogType.debug => logger.fine(message),
            LogType.error => logger.severe(message),
            LogType.info => logger.info(message)
          },
    );

File: src/application/commands/log.ts
import { ReaderTask } from 'fp-ts/ReaderTask'
import { Task } from 'fp-ts/Task'

export enum LogType {
	info,
	debug,
	error
}

export type Log = (type: LogType, message: string) => Task<void>

export type LogWithDeps<ENV> = (
	type: LogType,
	message: string
) => ReaderTask<ENV, void>
